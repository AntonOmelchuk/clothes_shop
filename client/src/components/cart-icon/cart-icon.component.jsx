import React from 'react';
import {connect} from 'react-redux';
import {createStructuredSelector} from 'reselect';

import './cart-icon.style.scss';

import {ReactComponent as ShoppingBagIcon} from '../../assets/shopping-bag.svg';

import {toggleHidden} from '../../redux/cart/cart.action';
import {selectCartItemsCount} from '../../redux/cart/cart.selectors';

const CartIcon = ({toggleHidden, itemsCount}) => (
  <div className='cart-icon' onClick={toggleHidden}>
    <ShoppingBagIcon className='shopping-icon' />
    <span className='item-count'>{itemsCount}</span>
  </div>
);

const mapStateToProps = createStructuredSelector({
  itemsCount: selectCartItemsCount,
});

export default connect(mapStateToProps, {toggleHidden})(CartIcon);
