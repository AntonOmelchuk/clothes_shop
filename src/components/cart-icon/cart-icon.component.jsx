import React from 'react';
import {connect} from 'react-redux';

import {ReactComponent as ShoppingBagIcon} from '../../assets/shopping-bag.svg';

import {toggleHidden} from '../../redux/cart/cart.action';
import {selectCartItemsCount} from '../../redux/cart/cart.selectors';

import './cart-icon.style.scss';

const CartIcon = ({toggleHidden, itemsCount}) => (
  <div className='cart-icon' onClick={toggleHidden}>
    <ShoppingBagIcon className='shopping-icon' />
    <span className='item-count'>{itemsCount}</span>
  </div>
);

const mapStateToProps = state => ({
  itemsCount: selectCartItemsCount(state),
});

export default connect(mapStateToProps, {toggleHidden})(CartIcon);
