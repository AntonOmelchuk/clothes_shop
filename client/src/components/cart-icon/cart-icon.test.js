import React from 'react';
import {shallow} from 'enzyme';
import {CartIcon} from './cart-icon.component';

let wrapper;
let mockToggleCartHidden;

beforeEach(() => {
  mockToggleCartHidden = jest.fn();

  const mockProps = {
    itemCount: 0,
    toggleHidden: mockToggleCartHidden,
  };

  wrapper = shallow(<CartIcon {...mockProps} />);
});

it('renders CartIcon component', () => {
  expect(wrapper).toMatchSnapshot();
});
