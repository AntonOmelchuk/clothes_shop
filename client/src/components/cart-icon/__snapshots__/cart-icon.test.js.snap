// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CartIcon component 1`] = `ShallowWrapper {}`;
